### ISSUE TITLE
Issue parsing groups that are in a space separated list string

### ISSUE THREAD

**Issue by Ariana-Hlavaty** [jenkinsci/jwt-auth-plugin#2]
### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  ```text
  Jenkins: 2.332.1
  OS: Linux - 4.14.252-195.483.amzn2.x86_64
  jwt-auth:0.2.0
  matrix-auth:2.6.11
  role-strategy:3.2.0
  ```

</details>

### What Operating System are you using (both controller, and any agents involved in the problem)?

Both controller and agent uses Linux - 4.14.252-195.483.amzn2.x86_64

### Reproduction steps

1. Set up JWT Auth settings
3. Add the groups to the role based or matrix auth plugin
4. Check the Authorities in <JENKINS_URL>/whoAmI

### Expected Results

- "authenticated"
- "220cd04c-aca6-405f-a95f-83d71d2ef62c"
- "a4aeb39e-e89b-4ffd-ad08-6e5fe3148254"

### Actual Results

- "authenticated"
- "[220cd04c-aca6-405f-a95f-83d71d2ef62c"
- " a4aeb39e-e89b-4ffd-ad08-6e5fe3148254]"

### Anything else?

The groups claim in the header contains the following string:
`"[220cd04c-aca6-405f-a95f-83d71d2ef62c a4aeb39e-e89b-4ffd-ad08-6e5fe3148254]"`

**Comment by narcoticfresh** [comment]
@Ariana-Hlavaty-i2 
sorry for the late reply..

it seems to me that the groups string is correctly split by " ".. your problem is that the enclosing `[`, `]` are messing up your group names/ids..

are you using a particular older version of pomerium maybe? i observed that the group names were wrongly serialized and contained those wrapping characters (which implies a list)..

please update whatever gives you this string and/or otherwise clean up the header..

**Comment by narcoticfresh** [comment]
closing, seems that the header is malformed..

**Comment by Ariana-Hlavaty** [comment]
Hi @narcoticfresh, I must have missed your first reply. I haven't modified the header and it is coming directly from Azure SAML

**Comment by pjdarton** [comment]
Hi @narcoticfresh 
I work with @Ariana-Hlavaty-i2 and I've done some investigation to make sure it's not "just a case of user error" and, as far as I can tell, it wasn't user error - our options have been constrained by the configuration options available to us.

We're using Microsoft's Azure Active Directory for authentication, hosting our Jenkins instance in AWS using AWS' Cognito to give it jwt.
We followed the instructions in https://aws.amazon.com/blogs/security/how-to-set-up-amazon-cognito-for-federated-authentication-using-azure-ad/ .
In our case:
 - we also told Azure to output a "jenkins-mail" field that was the user email converted to lower-case:
![image](https://user-images.githubusercontent.com/984537/184169326-b6a5dbd9-30fd-4c21-84ab-ee944234b374.png)
...although we haven't made any use of that yet.
 - we defined attributes on the AWS end to receive these attributes (and some we've not used):
![image](https://user-images.githubusercontent.com/984537/184169621-f8355e7c-738b-48d5-a229-1c8c62696ca1.png)
 - we told AWS to map the attributes across from the Azure SAML into AWS's Cognito:
![image](https://user-images.githubusercontent.com/984537/184169817-09477073-029e-454d-9d23-c280063d90ec.png)
   - SAML attribute http://schemas.microsoft.com/ws/2008/06/identity/claims/groups to a (custom String) User pool attribute `ad-group-name`,
   - to map the Azure SAML attribute `jenkins-mail` to a (custom String) User pool attribute of the same name, and
   - http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress to Email.
 - We've configured the jwt-auth plugin using the following JCasC:
```
jenkins:
  securityRealm:
    jwtAuth:
      allowVerificationFailures: true
      groupsClaimName: "custom:ad-group-name"
      headerName: "x-amzn-oidc-data"
      groupsClaimSeparator: ","
      leewaySeconds: 1
      userClaimName: "email"
```
Note that AWS attributes can only ever be Strings or Numbers in the AWS WebUI - there is no "List" option for an AWS attribute. The AWS CLI claims to support Strings, Numbers, timestamps and booleans, but the WebUI only supports Strings and Numbers, so we used "String".

When we authenticate, the jwt-auth plugin gets a `jwtClaims` object of the form:
```
JWT Claims Set:{sub=<redactedUuidA>, identities=[{"userId":"<redactedEmail>","providerName":"sso-dev-cognito-idp","providerType":"SAML","issuer":"https://sts.windows.net/<redactedUuidB>/","primary":true,"dateCreated":1649334747763}], email_verified=false, custom:ad-group-name=[<redactedUuidC>, <redactedUuidD>, <redactedUuidE>], custom:jenkins-mail=<redactedemailinlowercase>, email=<redactedEmail>, username=sso-dev-cognito-idp_<redactedEmail>, exp=1660217865, iss=https://cognito-idp.eu-west-2.amazonaws.com/eu-west-2_U1Qii8pJe}
```

With the 0.2.0 plugin, it correctly obtains the username BUT, when it comes to parsing the list of groups, it thinks it's a String, and then the plugin fails to parse that string in the way I need/expect.
 - with the default settings of (only) using commas as a groupsClaimSeparator, the code doesn't explode but sets the `groups` list to `[<redactedUuidC>`, ` <redactedUuidD>`, ` <redactedUuidE>]`, giving me groups I can't use because of the unwanted characters attached to those groups.
 - if I configure the plugin to use a groupsClaimSeparator of `, []` (commas, spaces and square brackets) then I'd _expect_ it to successfully split the string and give me the groups `<redactedUuidC>`, `<redactedUuidD>`, `<redactedUuidE>` ... but it actually fails with an `java.lang.IllegalArgumentException: A granted authority textual representation is required` (...and further examination revealed that this is because when it split the string into a list it used [splitPreserveAllTokens](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#splitPreserveAllTokens-java.lang.String-java.lang.String-) and hence got "empty tokens created by adjacent separators", which it then passed to `new SimpleGrantedAuthority(String)`, which rejects empty strings with an IllegalArgumentException).

However, if I edit the code and replace the call to [splitPreserveAllTokens](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#splitPreserveAllTokens-java.lang.String-java.lang.String-) with [split](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#split-java.lang.String-java.lang.String-) then the empty strings are omitted and everything works as expected (and this is the change that I've put in PR #4)

TL;DR: I don't think we're doing anything "imaginatively incorrect" (because I don't think we can!) ... and a simple code change makes it work for us.
